02_helper_functions/ImportPX4LogData.m

```diff
@@ -141,4 +145,115 @@
+    disp('INFO: Starting to import the csv data into matlab.')
+    tic;
+    param_fields = fieldnames(params);
+    
+    if numel(param_fields) == 0
+        error('No params specified in the setupParams() function.') 
+    end
+    
+    if exist(fullCSVName, 'file') == 2 % make sure file was created
+        try
+            % read in the data
+            opts = detectImportOptions(fullCSVName);
+            opts.EmptyLineRule = 'read';
+            csv_data = readtable(fullCSVName, opts,'ReadRowNames',true);
+            size_csv_data = size(csv_data);
+
+            % convert extra column strings to doubles and group numeric param
+            % properties
+            if size_csv_data(2) > 2
+                extra_cols = str2double(csv_data{:,3:size_csv_data(2)});
+            else
+                extra_cols = [];
+            end
+            csv_numeric_data = [csv_data{:,2}, extra_cols]; 
+
+            for idx_data = 1:2:size_csv_data(1) % skip timestamp rows
+
+                for idx_params = 1:numel(param_fields)
+
+                    % check if logged field is in defined param list
+                    if strcmpi(string(csv_data{idx_data, 1}), string(param_fields{idx_params}))
+
+                        % set field in parameter struct
+                        temp_params = csv_numeric_data(idx_data:idx_data+1,:);
+                        temp_params = temp_params(~isnan(temp_params));
+                        paramvector.(param_fields{idx_params}) = ...
+                            timeseries(temp_params(1,:), temp_params(2,:));
+
+                        % set logged
+                        params.(param_fields{idx_params}).logged = true;
+
+                    end
+                end
+            end
+        catch
+            disp(['Could not process the param: ' char(param_fields{idx_params})]);
+        end
+    else
+        error('Parameter CSV file does not exist: %s', fullCSVName);
+    end
+
+    time_csv_import = toc;
+    disp(['INFO: Importing the csv data to matlab took ' char(num2str(time_csv_import)) ' s.']);
+
+    % check that we have a nonempy paramvector
+    if (loadingMode~=1) && (loadingMode~=2)
+        if numel(fieldnames(paramvector)) == 0
+            error(['Empty paramvector: Converted the ulog file to csv and parsed it.' newline ...
+                'Does the logfile contain any parameters specified in the setupParams() function?'])
+        end
+    else
+        if numel(fieldnames(paramvector)) == 0
+            error(['Empty paramvector: Tried to read directly from the csv files.' newline ...
+                'Does a csv file containing parameters specified in the setupParams() function exist?'])
+        end
+    end
+
+    % *********************************
+    % save the paramvector and params struct if requested
+    % *********************************
+    if saveMatlabData
+        save([matLocation pathDelimiter fileName '.mat'], 'paramvector', 'params', '-append');
```

Comment:
As this assumes that the already exists why not saving the sysvector and the params at the same time? The same applies for deleting the csv files.
In that way we can remove similar code.

I was also quickly debating whether we should put the params into the sysvector to keep everything in one variable but I think your way is fine.

Reply:
Yes I also thought about putting both in the sysvector, but decided against it because I wanted some separation between topics, which have multiple members, and params. This could of course be debated, however, which is better. I'm open to opinions.